### PR TITLE
Add setup troubleshooting

### PR DESCRIPTION
**Pull request by ESKYoung**
Add How to guides for installation/setup issues, mainly for new
machines. Amend Dnsmasq manual setup instructions to reflect
error messages shown if file is out-of-date.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -238,7 +241,84 @@ govuk-docker prune
...
+### How to: resolve setup issues
+
+The following sections resolve common installation issues.
+
+#### How to: update Ruby
+
+Update Ruby using [brew](https://brew.sh/), then add it to the PATH in your bash config.
+
+```
+brew install ruby
```
Comment (issyl0) [inline README.md]: The preferred way is to use `rbenv` to manage Ruby versions. With `rbenv`, an `rbenv install` in the root of the repo will install the version of Ruby specified in [`.ruby-version`](https://github.com/alphagov/govuk-docker/blob/master/.ruby-version), which is probably different (ie older) than the latest provided by Homebrew.

File: README.md [hunk 2]
```diff
@@ -238,7 +241,84 @@ govuk-docker prune
...
+
+Install the correct version of bundler; you may have to overwrite existing bundle and bundler executable conflicts.
+
+```
+gem install bundler:X.Y.Z
+```
+
+#### How to: resolve issues caused by an existing docker install 
+
+During `govuk-docker setup`, if get the follow errors when pouring `docker-compose`:
```
Comment (issyl0) [inline README.md]:
```suggestion
During `govuk-docker setup`, if you get the following errors when pouring `docker-compose`:
```

File: README.md [hunk 3]
```diff
@@ -238,7 +241,84 @@ govuk-docker prune
...
+
+```
+Error: It seems there is already an App at '/Applications/Docker.app'.
+```
+
+Then uninstall your existing docker, and restart the `govuk-docker setup` process to install a new version of docker using brew.
+
+#### How to: resolve `No such file or directory` errors for `dev.gov.uk`
+
+If you get the following error during `govuk-docker setup`:
```
Comment (issyl0) [inline README.md]: There's a PR to fix this error fully at #160, but these docs are good for now. 👍

**Comment by ESKYoung** [comment]
Thanks for the useful feedback @issyl0 - I've made the changes you've suggested!

File: README.md [hunk 4]
```diff
@@ -245,16 +245,18 @@ You should then be able to `govuk-docker build` your service and have confidence
 
 The following sections resolve common installation issues.
 
-#### How to: update Ruby
+#### How to: install/update Ruby
 
-Update Ruby using [brew](https://brew.sh/), then add it to the PATH in your bash config.
+Follow the instructions to install [rbenv](https://github.com/rbenv/rbenv#installation) using [brew](https://brew.sh/).
```
Comment (issyl0) [inline README.md]: We can replace this sentence with `brew install rbenv` in the code block below, I think.

I'm tempted to also suggest adding it to the [Brewfile](https://github.com/alphagov/govuk-docker/blob/master/Brewfile), but that only runs on `govuk-docker setup` and if you're in this section then I assume that `govuk-docker` isn't running at all.

Comment (ESKYoung) [reply README.md]: There's a few additional steps after `brew install rbenv` to complete installation, including setting up `rbenv` in the shell with an added line for `.bash_profile`, and then running the rbenv-doctor script to validate install.

Can add these all to the code block if appropriate?

Comment (issyl0) [reply README.md]: Now you mention all those steps, I'm going to say let's stick with your original suggestion of following the `rbenv` README. We don't want to duplicate every instruction.

Comment (ESKYoung) [reply README.md]: Thanks - have fixed-up branch, and will merge in PR.

**Pull request merged**